aioftp/client.py

```diff
@@ -338,6 +341,188 @@ def parse_directory_response(self, s):
 
         return pathlib.PurePosixPath(directory)
 
+    @staticmethod
+    def parse_unix_mode(s):
+        """
+        Parsing unix mode strings("rwxr-x--t") into
+        hexacimal notation
+
+        :param s: mode string
+        :type s: :py:class:`str`
+
+        :return mode:
+        :rtype: :py:class:`int`
+        """
+        def parse_rw(s):
+            mode = 0
+
+            if s[0] == 'r':
+
+                mode |= 4
+
+            elif s[0] != '-':
+
+                raise ValueError
```

Comment:
In unix, the permissions can be either allowed or disallowed when reading is allowed, it should be 'r', if reading is not allowed, then it should be '-', if reading is not allowed and it is not a '-', then it means something is wrong(bug in our code; bug in the server, or it is an unknown format). If it is read only, it should be 'r--r--r--' if it is write only, then it should be '-w--w--w-', under no (unbuggy) circumstances should it be something other than 'r' or '-' in the read bit, and 'w' or '-' in the write bit